client/config.ts

```diff
@@ -103,7 +115,10 @@ export function configure(config?: Config) {
       crypto: config.crypto ?? Defaults.crypto,
       storage: config.storage ?? Defaults.storage,
       fetch: config.fetch ?? Defaults.fetch,
-      location: config.location ?? Defaults.location,
+      location: config.location ?? config.fido2?.passkeyDomain ? {
+        href: `https://${config.fido2?.passkeyDomain}`,
```

Comment:
@ottokruse why typescript still complains in this and next line that `config.fido2` could be undefined (forcing me to add the '?') when we know for sure it's never going to be undefined given the previous' line `config.fido2?.passkeyDomain ?`
Any hint on making TS a bit smarter and avoid the extra '?'